File: update_md.js
const fs = require('fs');
const path = require('path');

const dir = path.join(__dirname, 'src', 'content', 'services', 'spanish');
const files = fs.readdirSync(dir);
let count = 0;

files.forEach(file => {
  if (file.endsWith('.md') || file.endsWith('.mdx')) {
    const filePath = path.join(dir, file);
    let content = fs.readFileSync(filePath, 'utf-8');
    
    // Check if it already has soluciones_images
    if (content.includes('soluciones_images:')) {
      console.log(`Skipping ${file} - already has soluciones_images`);
      return;
    }

    // Find the image field
    const imageMatch = content.match(/image:\s*['"]([^'"]+)['"]/);
    if (imageMatch) {
      const mainImage = imageMatch[1];
      const lastDot = mainImage.lastIndexOf('.');
      if (lastDot !== -1) {
        const basePath = mainImage.substring(0, lastDot);
        const ext = mainImage.substring(lastDot);
        const image2 = `${basePath}2${ext}`;
        const image3 = `${basePath}3${ext}`;
        
        // Add soluciones_images right after the image field
        const replacement = `image: "${mainImage}"\nsoluciones_images:\n  - "${image2}"\n  - "${image3}"`;
        content = content.replace(/image:\s*['"]([^'"]+)['"]/, replacement);
        
        fs.writeFileSync(filePath, content, 'utf-8');
        console.log(`Updated ${file}`);
        count++;
      }
    } else {
        console.log(`No image field found in ${file}`);
    }
  }
});
console.log(`Updated ${count} files.`);

File: update_md.py
import os
import re

directory = os.path.join(os.getcwd(), 'src', 'content', 'services', 'spanish')
count = 0

for filename in os.listdir(directory):
    if filename.endswith(".md") or filename.endswith(".mdx"):
        filepath = os.path.join(directory, filename)
        with open(filepath, 'r', encoding='utf-8') as f:
            content = f.read()
            
        if 'soluciones_images:' in content:
            continue
            
        match = re.search(r'image:\s*[\'"]([^\'"]+)[\'"]', content)
        if match:
            main_image = match.group(1)
            base_path, ext = os.path.splitext(main_image)
            image2 = f"{base_path}2{ext}"
            image3 = f"{base_path}3{ext}"
            
            replacement = f'image: "{main_image}"\nsoluciones_images:\n  - "{image2}"\n  - "{image3}"'
            new_content = re.sub(r'image:\s*[\'"]([^\'"]+)[\'"]', replacement, content)
            
            with open(filepath, 'w', encoding='utf-8') as f:
                f.write(new_content)
                
            print(f"Updated {filename}")
            count += 1

print(f"Total updated: {count}")
